### ISSUE TITLE
Order by clause missing

### ISSUE THREAD

**Issue by Cybeer1111** [linq2db/linq2db#426]
Hi!

Here is the code:
```cs
(from oph in dataContext.GetEntityService<OperatorCallHistoryEntity>().List
                                orderby oph.StartDate descending
                                where oph.PhoneNumber == phoneNumber && !string.IsNullOrEmpty(oph.TransferPhoneNumber)
                                select oph.TransferPhoneNumber)
                                .Distinct()
                                .Select(e => new {Number = e})
                                .Take(10)
                                .Select(e => e.Number)
```
Here is generated SQL code:
```sql
SELECT FIRST 10
    t2.TRANSFERPHONENUMBER as TRANSFERPHONENUMBER1
FROM
    (
        SELECT DISTINCT
            t1.TRANSFERPHONENUMBER
        FROM
            OPERATORCALLHISTORY t1
        WHERE
            t1.PHONENUMBER = @phoneNumber AND NOT (t1.TRANSFERPHONENUMBER IS NULL OR Char_Length(t1.TRANSFERPHONENUMBER) = 0)
    ) t2
```
Order by clause is missing.

**Comment by sdanyliv** [comment]
Please write valid SQL that you expected after transferring such Linq query?

Looks like you need window functions which are currently not supported by linq2db and i think by other linq providers also.
https://www.simple-talk.com/sql/learn-sql-server/window-functions-in-sql-server/

**Comment by sdanyliv** [comment]
It is same as #437

**Comment by MaceWindu** [comment]
Fixed by #1221
